lib/matplotlib/backends/web_backend/js/mpl.js

```diff
@@ -642,29 +642,25 @@ mpl.figure.prototype._key_event_extra = function (_event, _name) {
 mpl.figure.prototype.key_event = function (event, name) {
     // Prevent repeat events
     if (name === 'key_press') {
-        if (event.which === this._key) {
+        if (event.key === this._key) {
             return;
         } else {
-            this._key = event.which;
+            this._key = event.key;
         }
     }
     if (name === 'key_release') {
         this._key = null;
     }
 
     var value = '';
-    if (event.ctrlKey && event.which !== 17) {
+    if (event.ctrlKey && event.key !== 'Control') {
         value += 'ctrl+';
     }
-    if (event.altKey && event.which !== 18) {
+    if (event.altKey && event.key !== 'Alt') {
         value += 'alt+';
     }
-    if (event.shiftKey && event.which !== 16) {
-        value += 'shift+';
-    }
```

Comment:
Is shift never added now?

Reply:
Not any more.

`_handle_key(key)` was the one that required the `shift+` string on the key value. If `shift+` was on a key value, then the function removed the `shift+` of the string and depending on the remaining of the key code, it got the "correct" value (it was correct only if the user had a QWERTY keyboard and a US keyboard layout). e.g. "shift+2" was replaced with "@".

As `KeyboardEvent.key` handles that for us (even better, with support for non US keyboard layouts), we no longer need to specify the `shift+` modifier on a key value.